tests/virtctl/vnc.go

```diff
@@ -51,129 +49,73 @@ import (
+			)()
 			Expect(err).ToNot(HaveOccurred())
-			// According to the video device type to set the expected resolution
-			// The default resolution is 720x400 for vga device while it is 1280x800 for virtio-gpu device
-			xres, yres := getResolution(domain)
-			// Sometimes we can see initially a 640x480 resolution if we connect very early
-			By("gathering screenshots until we are past the first boot screen and see the expected 720x400 resolution")
-			Eventually(func() image.Image {
-				cmd := clientcmd.NewVirtctlCommand("vnc", "screenshot", "--namespace", vmi.Namespace, "--file", filePath, vmi.Name)
-				Expect(cmd.Execute()).To(Succeed())
-
-				f, err := os.Open(filePath)
-				Expect(err).ToNot(HaveOccurred())
-				defer f.Close()
-
-				img, err := png.Decode(f)
-				Expect(err).ToNot(HaveOccurred())
-				return img
-			}, 10*time.Second).Should(HaveResolution(xres, yres))
-		})
+		}()
+
+		Eventually(func(g Gomega) {
+			conn, err := net.Dial("tcp", fmt.Sprintf("127.0.0.1:%s", testPort))
+			g.Expect(err).ToNot(HaveOccurred())
+			defer conn.Close()
+
+			client, err := vnc.Client(conn, &vnc.ClientConfig{
+				ServerMessageCh: make(chan vnc.ServerMessage),
+				ServerMessages:  []vnc.ServerMessage{new(vnc.FramebufferUpdateMessage)},
+			})
+			g.Expect(err).ToNot(HaveOccurred())
+			defer client.Close()
+
+			g.Expect(client.DesktopName).To(ContainSubstring(vmi.Name))
+		}, 60*time.Second).Should(Succeed())
 	})
-})
 
-func getResolution(domain *launcherApi.DomainSpec) (X, Y int) {
-	videoType := domain.Devices.Video[0].Model.Type
-	if videoType == "virtio" {
-		X = 1280
-		Y = 800
-	} else {
-		X = 720
-		Y = 400
-	}
-	return X, Y
-}
-
-type ResolutionMatcher struct {
-	X, Y int
-}
-
-func (h ResolutionMatcher) Match(actual interface{}) (success bool, err error) {
-	x, y, err := imgSize(actual)
-	if err != nil {
-		return false, nil
-	}
-	return x == h.X && y == h.Y, nil
-}
-
-func (h ResolutionMatcher) FailureMessage(actual interface{}) (message string) {
-	x, y, err := imgSize(actual)
-	if err != nil {
-		return err.Error()
-	}
-	return fmt.Sprintf("Expected (X: %d, Y: %d) to match (X: %d, Y: %d)", x, y, h.X, h.Y)
-}
-
-func (h ResolutionMatcher) NegatedFailureMessage(actual interface{}) (message string) {
-	x, y, err := imgSize(actual)
-	if err != nil {
-		return err.Error()
-	}
-	return fmt.Sprintf("Expected (X: %d, Y: %d) to not match (X: %d, Y: %d)", x, y, h.X, h.Y)
-}
-
-func imgSize(actual interface{}) (X, Y int, err error) {
-	if actual == nil {
-		return -1, -1, fmt.Errorf("expected an object of type image.Image but got nil")
-	}
-	img, ok := actual.(image.Image)
-	if !ok {
-		return -1, -1, fmt.Errorf("expected an object of type image.Image")
-	}
-	size := img.Bounds().Size()
-	return size.X, size.Y, nil
-}
-
-func HaveResolution(X, Y int) ResolutionMatcher {
-	return ResolutionMatcher{X: X, Y: Y}
-}
+	It("should allow creating a VNC screenshot in PNG format", func() {
+		domain, err := tests.GetRunningVMIDomainSpec(vmi)
```

Comment:
Improvement to the future, we should not need the domain but rather find out what type is expected based on the VMI

Reply:
Addressed, thanks for the hint!